### PR TITLE
SDK 247 - Add router from balancer to curve & vice versa

### PR DESCRIPTION
**Pull request by Preston-Harrison**

### CODE REVIEW

File: contracts/RouterBalancerCurve.sol [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import "./utils/CurveUtils.sol";
+import "./utils/BalancerUtils.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title RouterBalancerCurve
+ * @notice swaps tokens through balancer and curve pools
+ */
+contract RouterBalancerCurve is BaseRouter, CurveUtils, BalancerUtils {
```
Comment (rellfy) [inline contracts/RouterBalancerCurve.sol:15]: Could you add the gorilla comment here? :)

File: contracts/utils/BalancerUtils.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        );
+
+        // approve and perform swap
+        IERC20(from).approve(balancerVault, amount);
+        IBalancerVault(balancerVault).swap(
+            singleSwap,
+            fundManagement,
+            0, // min out not handled in this function, so it is set to zero
+            type(uint256).max // no deadline for this swap, hence deadline is infinite
+        );
```
Comment (rellfy) [inline contracts/utils/BalancerUtils.sol:57]: Please place the comments above the line
```suggestion
        // approve and perform swap
        IERC20(from).approve(balancerVault, amount);
        IBalancerVault(balancerVault).swap(
            singleSwap,
            fundManagement,
            // min out not handled in this function, so it is set to zero
            0,
            // no deadline for this swap, hence deadline is infinite
            type(uint256).max
        );
```

File: tests/unit/RouterBalancerCurve.spec.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  let MockMetapoolFactoryContract: MockMetapoolFactory;
+  let MockBalancerVault: MockBalancerVault;
+  let RouterBalancerCurve: RouterBalancerCurve;
+
+  before(async function () {
+    // await hre.network.provider.request({
+    //   method: "hardhat_reset",
+    //   params: []
+    // });
+
```
Comment (rellfy) [inline tests/unit/RouterBalancerCurve.spec.ts:33]:
```suggestion
```

**Review by rellfy — COMMENTED** [review]
Looks good, I just have some formatting comments

File: contracts/RouterBalancerCurve.sol (hunk 1 above)
Comment (Preston-Harrison) [reply contracts/RouterBalancerCurve.sol:15]: 🦍

File: tests/unit/RouterBalancerCurve.spec.ts (hunk 3 above)
Comment (Preston-Harrison) [reply tests/unit/RouterBalancerCurve.spec.ts]: done

File: contracts/utils/BalancerUtils.sol (hunk 2 above)
Comment (Preston-Harrison) [reply contracts/utils/BalancerUtils.sol]: done

File: tests/unit/RouterBalancerCurve.spec.ts (hunk 3 above)
Comment (Preston-Harrison) [reply tests/unit/RouterBalancerCurve.spec.ts]: done

**Review by brettvniekerk — APPROVED** [review]
LGTM

**Review by rellfy — APPROVED** [review]

**Pull request merged**